topics/assembly/tutorials/assembly-decontamination/tutorial.md

```diff
@@ -0,0 +1,324 @@
+>    - *"SED Program"*: `s/[a-z]/N/g`
+>
+>
+{: .hands_on}
+
+
+# Identify Non-Target Contaminants
+
+[Kraken2](https://ccb.jhu.edu/software/kraken2/) is a tool for taxonomic classification. It matches the kmer found in our seemblies to the ones stored in its databases. For the purpose of decontamination, we only need to identify possible contaminants, not the sequences belonging to our sampled organisms, we therefore only need to use a database containing likely contaminants such as bacteria, virus, and human (always wear gloves folks!).
+The best option at the time of this tutorial's writing is the PlusPF database which contains the RefSeq Standard (archaea, bacteria, viral, plasmid, human, UniVec_Core) plus protozoa and fungi data.
+
+
+
+> <hands-on-title> Identify the taxonomy of our sequences </hands-on-title>
+>
+> 1. {% tool [Kraken2](toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.1.1+galaxy1) %} with the following parameters:
+>    - *"Single or paired reads"*: `Single`
+>        - {% icon param-file %} *"Input sequences"*: `output` (output of **Text transformation** {% icon tool %})
+>    - *"Print scientific names instead of just taxids"*: `Yes`
+>    - *"Confidence"*: `0.3`
+>    - *"Split classified and unclassified outputs?"*: `Yes`
+>    - *"Select a Kraken2 database"*: ``
+>
+>    ***TODO***: *Check parameter descriptions*
+>
+>    ***TODO***: *Consider adding a comment or tip box*
+>
+>
+{: .hands_on}
+
+
+Kraken provides us with three outputs :
+- A table containing the classification informations for each sequences
+- A fasta file containing the classified sequences (our contaminants)
+- A fasta file containing the unclassified sequences (our precious zebra finch)
+
+We could decide to use the fasta file of unclassified sequences as our assembly, but remember that our genome is hard masked. That means that we lost information. Because we don't use it now doesn't mean we never will. So instead of using the masked genome, we will remove the classified sequences from our original assembly.
+
+
+But first, we are naturally curious people, and we want to know what type of contaminants we had in our sample.
+
+
+## Investigate our contaminants
+
+> <hands-on-title> Extract classification informations </hands-on-title>
+>
+> 1. {% tool [Cut](Cut1) %} with the following parameters:
+>    - *"Cut columns"*: `c1,c2,c3`
+>    - *"Delimited by"*: `Tab`
+>    - {% icon param-file %} *"From"*: `Kraken2 on data X: Classification` (output of **Kraken2** {% icon tool %})
+>
+>
+>    > <comment-title> Columns definitions </comment-title>
+>    >
+>    > The three columns that we extracted contain:
+>    > 1. Wheter the sequence is classified `C` or unclassified `U`
+>    > 2. The sequence name
+>    > 3. The sequence classification
+>    >
+>    {: .comment}
+>
+>    > <comment-title> What if my organism is in the database? </comment-title>
```

Comment:
```suggestion
>    > <comment-title> What if my organism is not in the database? </comment-title>
```

(I think?)

Reply:
The problem appears if the organism we assembled is in the databas, because we only keep the sequences that are unclassified (database contains only possible contaminant, other species stay unclassified). I tried to make that more clear in the comment box

Reply:
ahh gotcha, ok cool :+1: